src/vcpkg/vcpkgpaths.cpp

```diff
@@ -548,7 +548,7 @@ namespace vcpkg
                         Debug::print("Lockfile value for key '", x.first, "' was not a git commit sha\n");
                         return ret;
                     }
-                    ret.lockdata.emplace(x.first.to_string(), LockFile::EntryData{sv.to_string(), true});
+                    ret.lockdata.emplace(RepoAndRef::parse(x.first), LockFile::EntryData{sv.to_string(), true});
```

Comment:
I'd prefer to introduce another layer of object than to smoosh the two fields together